src/components/product/product.service.ts

```diff
@@ -441,27 +401,39 @@ export class ProductService {
         value: securedProperties.purposes.value ?? [],
       },
       produces: {
+        ...(securedProperties.scriptureReferencesOverride
+          ? securedProperties.scriptureReferencesOverride
+          : {
+              canRead: false,
+              canEdit: false,
+            }),
         value: {
           ...producible,
-          id: produces.p.properties.id,
+          id: pr.p.properties.id,
           __typename: (ProducibleType as any)[typeName],
-          scriptureReferences: !scriptureReferencesOverride.length
+          scriptureReferences: !scriptureReferencesOverrideValue.length
             ? producible?.scriptureReferences
             : {
-                canRead: result.canScriptureReferencesOverrideRead,
-                canEdit: result.canScriptureReferencesOverrideEdit,
-                value: scriptureReferencesOverride,
+                ...(securedProperties.scriptureReferencesOverride
+                  ? securedProperties.scriptureReferencesOverride
+                  : {
+                      canRead: false,
+                      canEdit: false,
+                    }),
+                value: scriptureReferencesOverrideValue,
               },
         },
-        canRead: result.canProducesRead,
-        canEdit: result.canProducesEdit,
       },
       scriptureReferencesOverride: {
-        canRead: result.canScriptureReferencesOverrideRead,
-        canEdit: result.canScriptureReferencesOverrideEdit,
-        value: !scriptureReferencesOverride.length
+        ...(securedProperties.scriptureReferencesOverride
+          ? securedProperties.scriptureReferencesOverride
+          : {
+              canRead: false,
+              canEdit: false,
+            }),
+        value: !scriptureReferencesOverrideValue.length
           ? null
-          : scriptureReferencesOverride,
+          : scriptureReferencesOverrideValue,
```

Comment:
An empty list is not the same thing as null here. null means there's no override. empty list means there is an override of an empty list.

Reply:
Empty list from `listScriptureReferences` function means scriptureReferencesOverride does not exist(we are saying this is null)

Reply:
We need to make the distinction between an empty list and null. They mean two different things. Maybe we need another db property like `isOverriding` to help.

Reply:
It's great if we can use isOverriding to differentiate between override states instead of using null.